src/redis_set.cc

```diff
@@ -168,7 +168,7 @@ rocksdb::Status Set::Take(const Slice &user_key, std::vector<std::string> *membe
   std::string ns_key;
   AppendNamespacePrefix(user_key, &ns_key);
 
-  if (pop) LockGuard guard(storage_->GetLockManager(), ns_key);
```

Comment:
the intention of this line was locked when `pop` is true, but it should be never working since the lock would be released after exiting the `if` block. : (

Reply:
I think we can lock it, if !pop then unlock it immediately, this will have a little overhead.

Reply:
yup, the main issue was read operation may block which was not expected. it seems fine that use lock/unlock the key seperately.

Reply:
has filed another PR to fix this issue: https://github.com/bitleak/kvrocks/pull/256